src/secrets/src/lib.rs

```diff
@@ -83,7 +83,7 @@ impl SecretsReader {
     ///
     /// (N.B. Were we ever to run with Windows / NTFS, this would also work
     /// properly _and_ mid-read edits would be disallowed)
-    pub fn read(&self, id: GlobalId) -> Result<Vec<u8>, anyhow::Error> {
+    pub async fn read(&self, id: GlobalId) -> Result<Vec<u8>, anyhow::Error> {
```

Comment:
Is using this coming in a future change? like using tokio::io instead of std::io?

Reply:
Yessir! The next PR is going to swap in a `SecretsReader` trait object that either reads from the K8s API (which is mandatorily async) or the existing files.